src/future.cpp

```diff
@@ -140,154 +188,18 @@ class task_system {
         _q[i % _count].push(forward<F>(f));
     }
 
-    // TODO FP: delayed execution to be implemented
     template <typename F>
-    void async_(std::chrono::system_clock::time_point, F&& f) {
-        auto i = _index++;
-
-        for (unsigned n = 0; n != _count; ++n) {
-            if (_q[(i + n) % _count].try_push(forward<F>(f))) return;
-        }
-
-        _q[i % _count].push(forward<F>(f));
-    }
-};
-
-/**************************************************************************************************/
-
-#if 0
-struct timed_queue {
-    using lock_t = unique_lock<mutex>;
-    using element_t = pair<steady_clock::time_point, any_packaged_task_>;
-    using queue_t = vector<element_t>;
-
-    queue_t             _q;
-    condition_variable  _condition;
-    bool                _done = false;
-    mutex               _mutex;
-    thread              _thread;
-
-    timed_queue() { _thread = thread([this]{ this->_run(); }); }
-
-    ~timed_queue() {
-        {
-        lock_t lock(_mutex);
-        _done = true;
-        }
-        _thread.join();
-    }
-
-    void _run()
-    {
-        while(true) {
-            any_packaged_task_ task;
-            {
-            lock_t lock(_mutex);
-
-            while (_q.empty() && !_done) _condition.wait(lock);
-            if (_q.empty()) break; // ==> _done
-            while (steady_clock::now() < _q.front().first) {
-                auto t = _q.front().first;
-                _condition.wait_until(lock, t);
-            }
-            pop_heap(begin(_q), end(_q), [](const auto& x, const auto& y) {
-                return x.first > y.first;
-            });
-            task = move(_q.back().second);
-            _q.pop_back();
-            }
-
-            adobe::details::async_(move(task));
-        }
-    }
-
-    void async(const steady_clock::time_point& when, any_packaged_task_&& p)
-    {
+    void async_(std::chrono::system_clock::time_point when, F&& f) {
```

Comment:
Should check the time_point for time_point::min() and skip the timed queue and heap operations in that case.